src/data/services/JournalsApiService.js

```diff
@@ -30,6 +35,18 @@ class JournalsApiService {
       withCredentials: true,
     });
   }
+
+  static setPageVisit(userId, pageId) {
+    return axios.post(`${JournalsApiService.apiUrl}/userpagevisits/`, {
+      user: userId,
+      page: pageId,
+    }, {
+      withCredentials: true,
+      headers: {
+        'X-CSRFToken': getCookieValue('journals_csrftoken'),
```

Comment:
why do we need this for this api call only and not the others?

Reply:
Because POST calls are an "unsafe" call and only unsafe calls requires CSRF protection.